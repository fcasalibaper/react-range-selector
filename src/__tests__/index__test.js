/* eslint-disable */

import RangeSelector from '../index';
import renderer from 'react-test-renderer';
import React from 'react';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });


it('both coustomiseRange and range', () => {
  const tree = renderer.create(
    <RangeSelector componentName={'range-selector'}
    customiseRange={[2,5]}
    initalSelected={[3,4]}
    key={3}
    name={'car'}
    range={[3,5]}
  />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('not coustomiseRange or range', () => {
  const tree = renderer.create(
    <RangeSelector componentName={'range-selector'}
    initalSelected={[3,4]}
    key={3}
    name={'car'}
  />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders range', () => {
  const tree = renderer.create(
    <RangeSelector componentName={'range-selector'}
    key={3}
    name={'car'}
    range={[3,5]}
  />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders inital & range', () => {
  const tree = renderer.create(
    <RangeSelector componentName={'range-selector'}
    key={3}
    name={'car'}
    range={[3,5]}
    initalSelected={[0,2]}
  />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders customise', () => {
  const tree = renderer.create(
    <RangeSelector componentName={'range-selector'}
    key={3}
    name={'car'}
    customiseRange={[0.8,1.0,1.4,1.6,1.8,2.0,2.2]}
  />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders inital & customise', () => {
  const tree = renderer.create(
    <RangeSelector componentName={'range-selector'}
    key={3}
    name={'car'}
    customiseRange={[0.8,1.0,1.4,1.6,1.8,2.0,2.2]}
    initalSelected={[0,2]}
  />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('one point select render check', () => {
  const tree = shallow(<RangeSelector componentName={'range-selector'}
    initalSelected={[4]}
    key={3}
    name={'car'}
    range={[3,5]}
  />);
  expect(tree.instance().isInSelected(4)).toEqual(true);
});

it('two points select render check', () => {
  const tree = shallow(<RangeSelector componentName={'range-selector'}
    initalSelected={[2,4]}
    key={3}
    name={'car'}
    range={[3,7]}
  />);
  expect(tree.instance().isInSelected(4)).toEqual(true);
  expect(tree.instance().isInSelected(3)).toEqual(true);
  expect(tree.instance().isInSelected(2)).toEqual(true);
});