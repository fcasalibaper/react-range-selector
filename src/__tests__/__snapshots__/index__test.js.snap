// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`both coustomiseRange and range 1`] = `
<h1>
  Please use range or customiseRange, not both
</h1>
`;

exports[`not coustomiseRange or range 1`] = `
<h1>
  Please set range or customiseRange
</h1>
`;

exports[`renders customise 1`] = `
<div
  className="range-selector__item"
>
  <b
    className="range-selector__item-title"
  >
    car
  </b>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-0"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-0"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-0"
    >
      0.8
    </label>
  </div>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-1"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-1"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-1"
    >
      1
    </label>
  </div>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-2"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-2"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-2"
    >
      1.4
    </label>
  </div>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-3"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-3"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-3"
    >
      1.6
    </label>
  </div>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-4"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-4"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-4"
    >
      1.8
    </label>
  </div>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-5"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-5"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-5"
    >
      2
    </label>
  </div>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-6"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-6"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-6"
    >
      2.2
    </label>
  </div>
</div>
`;

exports[`renders inital & customise 1`] = `
<div
  className="range-selector__item"
>
  <b
    className="range-selector__item-title"
  >
    car
  </b>
  <div
    className="range-selector__item-input is-start is-in-range"
  >
    <input
      checked={true}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-0"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-0"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-0"
    >
      0.8
    </label>
  </div>
  <div
    className="range-selector__item-input is-in-range"
  >
    <input
      checked={true}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-1"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-1"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-1"
    >
      1
    </label>
  </div>
  <div
    className="range-selector__item-input is-end is-in-range"
  >
    <input
      checked={true}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-2"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-2"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-2"
    >
      1.4
    </label>
  </div>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-3"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-3"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-3"
    >
      1.6
    </label>
  </div>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-4"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-4"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-4"
    >
      1.8
    </label>
  </div>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-5"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-5"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-5"
    >
      2
    </label>
  </div>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-6"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-6"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-6"
    >
      2.2
    </label>
  </div>
</div>
`;

exports[`renders inital & range 1`] = `
<div
  className="range-selector__item"
>
  <b
    className="range-selector__item-title"
  >
    car
  </b>
  <div
    className="range-selector__item-input is-start is-in-range"
  >
    <input
      checked={true}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-0"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-0"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-0"
    >
      3
    </label>
  </div>
  <div
    className="range-selector__item-input is-in-range"
  >
    <input
      checked={true}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-1"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-1"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-1"
    >
      4
    </label>
  </div>
  <div
    className="range-selector__item-input is-end is-in-range"
  >
    <input
      checked={true}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-2"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-2"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-2"
    >
      5
    </label>
  </div>
</div>
`;

exports[`renders range 1`] = `
<div
  className="range-selector__item"
>
  <b
    className="range-selector__item-title"
  >
    car
  </b>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-0"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-0"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-0"
    >
      3
    </label>
  </div>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-1"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-1"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-1"
    >
      4
    </label>
  </div>
  <div
    className="range-selector__item-input "
  >
    <input
      checked={false}
      className="range-selector__item-input-checkbox"
      id="range-selector_car-2"
      onChange={[Function]}
      type="checkbox"
      value="range-selector_car-2"
    />
    <label
      className="range-selector__item-input-text"
      htmlFor="range-selector_car-2"
    >
      5
    </label>
  </div>
</div>
`;
